### PR TITLE
OTT-652: Added changes for reading adunit from imp.tagID

### PR DESCRIPTION
**Pull request by PubMatic-OpenWrap**
# Description

Please add change description or link to ticket, docs, etc.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

File: floors/rule.go [hunk 1]
```diff
@@ -281,12 +281,36 @@ func extractChanelNameFromBidRequestExt(bidRequest *openrtb2.BidRequest) string
...
+
+func getAdUnitCode(imp openrtb2.Imp) string {
+	adUnitCode := CATCH_ALL
+	gpId, err := jsonparser.GetString(imp.Ext, "gpid")
+	if err == nil && gpId != "" {
+		return gpId
+	}
+
+	tagID := imp.TagID
+	if tagID != "" {
```
Comment (pm-nilesh-chate) [inline floors/rule.go:300]: Is new variable `tagID` required?

File: floors/rule.go [hunk 2]
```diff
@@ -281,12 +281,36 @@ func extractChanelNameFromBidRequestExt(bidRequest *openrtb2.BidRequest) string
...
+		return tagID
+	}
+
 	pbAdSlot, err := jsonparser.GetString(imp.Ext, "data", "pbadslot")
-	if err == nil {
-		value = pbAdSlot
+	if err == nil && pbAdSlot != "" {
```
Comment (pm-nilesh-chate) [inline floors/rule.go:305]: I would recommend to cover these new changes in unit tests

File: floors/rule.go (hunk 1 above)
Comment (PubMatic-OpenWrap) [reply floors/rule.go:300]: Done

File: floors/rule.go (hunk 2 above)
Comment (PubMatic-OpenWrap) [reply floors/rule.go:304]: Done

**Review by pm-viral-vala — APPROVED** [review]

**Pull request merged**